openvdb_ax/openvdb_ax/compiler/CompilerOptions.h

```diff
@@ -0,0 +1,74 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+
+/// @file compiler/CompilerOptions.h
+///
+/// @authors Nick Avramoussis
+///
+/// @brief  OpenVDB AX Compiler Options
+///
+
+#ifndef OPENVDB_AX_COMPILER_COMPILER_OPTIONS_HAS_BEEN_INCLUDED
+#define OPENVDB_AX_COMPILER_COMPILER_OPTIONS_HAS_BEEN_INCLUDED
+
+#include <openvdb/openvdb.h>
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+
+namespace ax {
+
+/// @brief Options that control how functions behave
+struct FunctionOptions
+{
+    /// @brief  Enable the constant folding of C bindings. Functions may use this setting
+    ///         to determine whether they are allowed to be called during code generation
+    ///         to evaluate call sites with purely constant arguments and replace the call
+    ///         with the result.
+    /// @note   This does not impact IR functions which we leave to LLVM's CF during
+    ///         IR optimization.
+    /// @note   We used to bind IR methods to corresponding C bindings, however it can be
+    ///         very easy to implement incorrectly, leading to discrepancies in the CF
+    ///         results. Fundamentally, LLVM's support for CF IR is far superior and our
+    ///         framework only supports some types of folding (see codegen/ConstantFolding.h)
+    bool mConstantFoldCBindings = true;
```

Comment:
I tend to think of m prefixed variables as being private or protected members. For these type of structs, I would suggest just including the name itself so you don't need the user to include the m prefix or otherwise provide getters and setters, but I don't think that's necessary here.